### DISCUSSION TITLE
failed assertion with 'Texture Descriptor Validation' on macOS 13.0

### DISCUSSION THREAD

**Discussion by floriangbh** [HaishinKit/HaishinKit.swift#1181] [Q&A] [answered]
### Describe the bug

Hello, 

Since the MacOS Ventura update we are not unable to run any stream due to a Metal assertion : 

```
-[MTLDebugDevice newTextureWithDescriptor:iosurface:plane:]:2403: failed assertion `Texture Descriptor Validation
IOSurface textures must use MTLStorageModeShared
```

I find this on the developers forums : https://developer.apple.com/forums/thread/710843

It look like a bug with MacOS Ventura 13.0, but if someone have any solution I'm taking ! 

### To Reproduce

1. Run the "Exemple  iOS+SwiftUI" exemple with "My Mac (designed for iPad)"
2. Accept permissions 
3. Restart and see the assertion

### Expected behavior

Working on Silicon device with (Designed for iPad)

### Version

- MacOS 13.0
- HaishinKit.swift 1.0

### Screenshots

<img width="1139" alt="Screenshot 2022-11-05 at 10 51 45" src="https://user-images.githubusercontent.com/14054997/200113916-2a4c8fbe-5374-4cf0-af23-105fe77e3663.png">

**Comment by shogo4405** [comment]
I had same issue with macOS 13.0. This workaround works for me.
Edit Scheme -> Run -> Diagnostics -> Metal API Validation -> Turn it off.
<img width="919" alt="スクリーンショット 2022-12-03 16 21 41" src="https://user-images.githubusercontent.com/810189/205429859-a7dd2224-7986-4d7e-924e-e1b16d302e5c.png">

**Comment by floriangbh** [comment]
Hello @shogo4405, 

Thanks for the workaround in debug mode, hope Apple will fix it for production...

**Comment by shogo4405** [comment]
I updated macOS 13.1. I had an another issue. `AVCaptureSession#startRunning` caused crash.
<img width="1261" alt="スクリーンショット 2022-12-18 15 44 15" src="https://user-images.githubusercontent.com/810189/208285353-ea85e970-d169-48e2-a248-9eb0cd616bfa.png">

Apple's exsample code had an same issue.
https://developer.apple.com/documentation/avfoundation/capture_setup/avmulticampip_capturing_from_multiple_cameras

**Comment by floriangbh** [comment]
I've added the sample code exemple in the Feedback Assistant

**Comment by shogo4405** [comment]
I updated macOS 13.2. 
- It seems that this fixed issue. https://github.com/shogo4405/HaishinKit.swift/issues/1080#issuecomment-1356699528

But this issue not fixed yet.  😭

**Comment by shogo4405** [comment]
Also Apple's [sample code](https://developer.apple.com/documentation/avfoundation/capture_setup/avmulticampip_capturing_from_multiple_cameras) crash on macOS 13.2.
<img width="1241" alt="スクリーンショット 2023-01-28 15 10 39" src="https://user-images.githubusercontent.com/810189/215250049-fe1290f7-bb0c-425e-90ec-15af7d5e2298.png">

**Comment by shogo4405** [comment]
I turn it off **Metal API Validation**.
<img width="925" alt="スクリーンショット 2023-01-28 15 15 52" src="https://user-images.githubusercontent.com/810189/215250201-f0b32a2d-a93a-4fde-add4-69f08b2da1f2.png">

**Comment by floriangbh** [comment]
Since the beginning of November I have had no response on the Apple feedback assistant. API on MacOS seems completely abandoned...

**Comment by shogo4405** [comment]
Also macOS Ventura 13.3.1. Would Apple be willing to fix it... I will move to GitHub Discussions.

**Comment by floriangbh** [comment]
Quick update from Apple on my feedback : 

<img width="762" alt="Screenshot 2023-06-17 at 15 42 45" src="https://github.com/shogo4405/HaishinKit.swift/assets/14054997/4d4b22d5-f63e-423e-9906-3afba1e07e07">

I don't have any extra Macbook to install the beta, I will get it in the next month to try.

**Comment by floriangbh** [comment, marked as answer]
Fixed on MacOS Sonoma Beta 5 !
